src/coffeelint.coffee

```diff
@@ -310,9 +309,10 @@ coffeelint.lint = (source, userConfig = {}, literate = false) ->
 
     # only do further checks if the syntax is okay, otherwise they just fail
     # with syntax error exceptions
-    unless hasSyntaxError(source)
+    tokens = getTokens(source)
```

Comment:
Here we were passing the source to Coffeescript before twice, which basically means we're doing twice as much work as needed.